src/XrmMockupShared/Utility.cs

```diff
@@ -583,8 +583,8 @@ private static bool EvaluateCondition(Entity row, ConditionExpression condition)
                     break;
             }
 
-            if (attr is EntityReference entityRef) { attr = entityRef.Id; }
             attr = ConvertToComparableObject(attr);
+            if (attr is EntityReference entityRef) { attr = entityRef.Id; }
```

Comment:
Is this check necessary? ConvertToComparableObject already returns entityReference.Id if attr is EntityReference.